gemrb/core/DisplayMessage.cpp

```diff
@@ -188,20 +188,22 @@ void DisplayMessage::DisplayString(String text, const Color &color, Scriptable *
 	}
 }
 
-std::map<int, std::string> DisplayMessage::GetAllColors() const
+std::map<GUIColors, std::string> DisplayMessage::GetAllColors() const
 {
-	std::map<int, std::string> auxiliaryColors;
+	std::map<GUIColors, std::string> auxiliaryColors;
 	AutoTable colorTable = gamedata->LoadTable("colors", true);
 	assert(colorTable);
 	for (TableMgr::index_t r = 0; r < colorTable->GetRowCount(); r++) {
-		auxiliaryColors[static_cast<int>(r)] = colorTable->GetRowName(r);
+		if (static_cast<int>(r) >= static_cast<int>(GUIColors::WHITE) && static_cast<int>(r) <= static_cast<int>(GUIColors::DIALOGPARTY)) {
```

Comment:
You can cast to one of the types already used, avoiding the need for so many casts.
But also this is a bit fragile:
- you can move these bounds conditions to the loop definition
- it would be cleaner if you now added two more entries to the enum, to mark the start and the end, so these bounds here won't need to be updated every time a new color is added. I suggest FIRST_COLOR and LAST_COLOR